core/domain/exp_domain.py

```diff
@@ -4224,6 +4222,9 @@ def validate(self):
             raise Exception(
                 'Assigned state ids should be smaller than last state id used.')
 
+        if not all(isinstance(x, int) for x in state_ids):
+            raise Exception('Assigned state ids should be integer values.')
```

Comment:
Why wouldn't this be the case?

Reply:
I have no idea but last time when I checked production models I encountered `null` values being set for END states.

Reply:
That is a good idea. I will modify this in next commit.

Reply:
Done